### PR TITLE
gh-110875: Handle '.' properties in logging formatter configuration c…

### PR DESCRIPTION
**Pull request by vsajip**
…orrectly.

* Issue: gh-110875

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Not sure I'm the best reviewer here -- not very familiar with this code, and haven't used `logging.Formatter` much. But I can try to take a look tomorrow :)

**Comment by vsajip** [comment]
It's a fairly simple localised change :smile:

File: Lib/logging/config.py [hunk 1]
```diff
@@ -482,10 +482,10 @@ def configure_custom(self, config):
         c = config.pop('()')
         if not callable(c):
             c = self.resolve(c)
-        props = config.pop('.', None)
         # Check for valid identifiers
-        kwargs = {k: config[k] for k in config if valid_ident(k)}
+        kwargs = {k: config[k] for k in config if (False if k == '.' else valid_ident(k))}
```
Comment (encukou) [inline Lib/logging/config.py]: Style nitpick:
```suggestion
        kwargs = {k: config[k] for k in config if k != '.' and valid_ident(k)}
```

**Review by encukou — COMMENTED** [review]
The same logic is in `DictConfigurator.configure_handler` as well. Does that need a fix too?

**Comment by vsajip** [comment]
> The same logic is in `DictConfigurator.configure_handler` as well. Does that need a fix too?

I don't think so, because the `.` key is popped, if present, just before the equivalent list comprehension:
https://github.com/python/cpython/blob/20cfab903db70cf952128bc6b606e3ec4a216498/Lib/logging/config.py#L838-L839

**Comment by encukou** [comment]
Well, that's how it was in `configure_custom` too, before this PR.

**Review by encukou — APPROVED** [review]
PR looks good to me.
If the other site needs to be fixed, it can be done in a different PR. Or this one if you prefer :)

**Comment by vsajip** [comment]
> If the other site needs to be fixed, it can be done in a different PR. Or this one if you prefer :)

I think you're right. Better to make the logic consistent in both places, and it also improves the locality of behavior re. usage of `props`. Thanks very much for the review!

**Pull request merged**